openstack/autoscaling/v1/lifecycle_hooks/Create.go

```diff
@@ -0,0 +1,59 @@
+package lifecyclehooks
+
+import (
+	golangsdk "github.com/opentelekomcloud/gophertelekomcloud"
+	"github.com/opentelekomcloud/gophertelekomcloud/internal/build"
+	"github.com/opentelekomcloud/gophertelekomcloud/internal/extract"
+)
+
+// CreateOpts represents the request parameter struct for lifecycle hook.
+type CreateOpts struct {
+	// Specifies the lifecycle hook name. The name contains only letters, digits, underscores (_), and hyphens (-), and cannot exceed 32 characters.
+	LifecycleHookName string `json:"lifecycle_hook_name" required:"true"`
+	// Specifies the lifecycle hook type. Options:
+	//   - INSTANCE_TERMINATING: The hook suspends the instance when it is terminated.
+	//   - INSTANCE_LAUNCHING: The hook suspends the instance when it is started.
+	LifecycleHookType string `json:"lifecycle_hook_type" required:"true"`
+	// Specifies the default lifecycle hook callback operation. This operation is performed when the timeout duration expires.
+	// Options: ABANDON(Default), CONTINUE
+	// ABANDON:
+	// If an instance is starting, ABANDON indicates that your customized operations failed, and the instance will be terminated.
+	// In such a case, the scaling action fails, and you must create a new instance.
+	// If an instance is stopping, ABANDON allows instance termination BUT stops other lifecycle hooks.
+	// CONTINUE:
+	// If an instance is starting, CONTINUE indicates that your customized operations are successful and the instance can be used.
+	// If an instance is stopping, CONTINUE allows instance termination AND the completion of other lifecycle hooks.
+	DefaultResult string `json:"default_result,omitempty"`
+	// Specifies the lifecycle hook timeout duration, which ranges from 60 to 86400 seconds. The default value is 3600.
+	DefaultTimeout int `json:"default_timeout,omitempty"`
```

Comment:
use here pointer, without pointer impossible to omit this parameter, could lead to error

Reply:
default values in api